lewis_emulators/attocube_anc350/interfaces/stream_interface.py

```diff
@@ -0,0 +1,149 @@
+from lewis.adapters.stream import StreamInterface, Cmd
+from lewis.core.logging import has_log
+from lewis_emulators.utils.byte_conversions import int_to_raw_bytes, raw_bytes_to_int
+from functools import partial
+
+BYTES_IN_INT = 4
+HEADER_LENGTH = 4*BYTES_IN_INT
+
+convert_to_response = partial(int_to_raw_bytes, length=BYTES_IN_INT, low_byte_first=True)
+
+# NB, all variables used from here onwards are named the same in the C driver
+# 'hump' in the controller refers to hardware limits
+UC_SET = 0  # Set command
+UC_GET = 1  # Get command
+UC_ACK = 3  # Ack command
+UC_TELL = 4  # Event command
+
+UC_REASON_OK = 0  # All ok
+UC_REASON_ADDR = 1  # Invalid address
+UC_REASON_RANGE = 2  # Value out of range
+UC_REASON_IGNORED = 3  # Telegram was ignored
+UC_REASON_VERIFY = 4  # Verify of data failed
+UC_REASON_TYPE = 5  # Wrong type of data
+UC_REASON_UNKNW = 99  # unknown error
+
+#  Memory addresses
+#  Read
+ID_ANC_AMPL = 0x400
+ID_ANC_FAST_FREQ = 0x0401
+ID_ANC_STATUS = 0x0404
+ID_ANC_REFCOUNTER = 0x0407  # The position of home
+ID_ANC_COUNTER = 0x0415
+ID_ANC_UNIT = 0x041D
+ID_ANC_SENSOR_VOLT = 0x0526
+ID_ANC_REGSPD_SETP = 0x0542  # Speed of the controller
+ID_ANC_REGSPD_SETPS = 0x0549  # Step width of the controller
+ID_ANC_MAX_AMP = 0x054F
+ID_ANC_CAP_VALUE = 0x0569
+
+#  Write
+ID_ANC_STOP_EN = 0x0450  # Enables 'hump' detection
+ID_ANC_REGSPD_SELSP = 0x054A  # Type of setpoint for the speed
+ID_ANC_TARGET = 0x0408  # The target to move to
+ID_ANC_RUN_TARGET = 0x040d  # Actually start the move to target
+
+
+# Status bitmask
+ANC_STATUS_RUNNING = 0x0001
+ANC_STATUS_HUMP = 0x0002
+ANC_STATUS_SENS_ERR = 0x0100
+ANC_STATUS_DISCONN = 0x0400
+ANC_STATUS_REF_VALID = 0x0800
+ANC_STATUS_ENABLE = 0x1000
+
+
+def convert_to_ints(command, start, end):
+    return [raw_bytes_to_int(command[x:x + BYTES_IN_INT]) for x in range(start, end, BYTES_IN_INT)]
+
+
+def generate_response(address, index, correlation_num, data=None):
+    int_responses = [UC_ACK, address, index, correlation_num, UC_REASON_OK]
+    if data is not None:
+        int_responses.append(data)
+    response = "".join(convert_to_response(x) for x in int_responses)
+    return convert_to_response(len(response)) + response
```

Comment:
Maybe a docstring on this method and the one above, as they are slightly "tricky"?